libusb/libusb.c

```diff
@@ -252,60 +507,220 @@ static usbd_respond cdc_control(usbd_device *dev, usbd_ctlreq *req, usbd_rqc_cal
+ */
+static void comm_rxtx(usbd_device *dev, uint8_t event, uint8_t ep) {
     if (event == usbd_evt_eptx) {
-        cdc_txonly(dev, event, ep);
-    } else {
-        cdc_rxonly(dev, event, ep);
+        comm_txonly(dev, ep);
+    } else if (event == usbd_evt_eprx){
+        comm_rxonly(dev, ep);
     }
 }
 
-static usbd_respond cdc_setconf (usbd_device *dev, uint8_t cfg) {
+static usbd_respond device_setconf (usbd_device *dev, uint8_t cfg) {
     switch (cfg) {
     case 0:
         /* deconfiguring device */
-        usbd_ep_deconfig(dev, CDC_NTF_EP);
-        usbd_ep_deconfig(dev, CDC_TXD_EP);
-        usbd_ep_deconfig(dev, CDC_RXD_EP);
-        usbd_reg_endpoint(dev, CDC_RXD_EP, 0);
-        usbd_reg_endpoint(dev, CDC_TXD_EP, 0);
+        usbd_ep_deconfig(dev, DEVICE_DATA_RX_EP);
+        usbd_ep_deconfig(dev, DEVICE_DATA_TX_EP);
+        usbd_reg_endpoint(dev, DEVICE_DATA_RX_EP, 0);
+        usbd_reg_endpoint(dev, DEVICE_DATA_TX_EP, 0);
+#if (ENABLE_HID_WEBUSB_COMM == 1)
+        usbd_ep_deconfig(dev, WEBUSB_RXD_EP);
+        usbd_ep_deconfig(dev, WEBUSB_TXD_EP);
+        usbd_reg_endpoint(dev, WEBUSB_RXD_EP, 0);
+        usbd_reg_endpoint(dev, WEBUSB_TXD_EP, 0);
+#endif
         return usbd_ack;
     case 1:
         /* configuring device */
-        usbd_ep_config(dev, CDC_RXD_EP, USB_EPTYPE_BULK /*| USB_EPTYPE_DBLBUF*/, CDC_DATA_SZ);
-        usbd_ep_config(dev, CDC_TXD_EP, USB_EPTYPE_BULK /*| USB_EPTYPE_DBLBUF*/, CDC_DATA_SZ);
-        usbd_ep_config(dev, CDC_NTF_EP, USB_EPTYPE_INTERRUPT, CDC_NTF_SZ);
-        usbd_reg_endpoint(dev, CDC_RXD_EP, cdc_rxtx);
-        usbd_reg_endpoint(dev, CDC_TXD_EP, cdc_rxtx);
-        usbd_ep_write(dev, CDC_TXD_EP, 0, 0);
+        usbd_ep_config(dev, DEVICE_DATA_RX_EP, DEVICE_DATA_EP_TYPE /*| USB_EPTYPE_DBLBUF*/, DEVICE_DATA_EP_SIZE);
+        usbd_ep_config(dev, DEVICE_DATA_TX_EP, DEVICE_DATA_EP_TYPE /*| USB_EPTYPE_DBLBUF*/, DEVICE_DATA_EP_SIZE);
+        usbd_reg_endpoint(dev, DEVICE_DATA_RX_EP, comm_rxtx);
+        usbd_reg_endpoint(dev, DEVICE_DATA_TX_EP, comm_rxtx);
+        usbd_ep_write(dev, DEVICE_DATA_TX_EP, 0, 0);
+#if (ENABLE_HID_WEBUSB_COMM == 1)
+        usbd_ep_config(dev, WEBUSB_RXD_EP, USB_EPTYPE_INTERRUPT /*| USB_EPTYPE_DBLBUF*/, DEVICE_DATA_EP_SIZE);
+        usbd_ep_config(dev, WEBUSB_TXD_EP, USB_EPTYPE_INTERRUPT /*| USB_EPTYPE_DBLBUF*/, DEVICE_DATA_EP_SIZE);
+        usbd_reg_endpoint(dev, WEBUSB_RXD_EP, comm_rxtx);
+        usbd_reg_endpoint(dev, WEBUSB_TXD_EP, comm_rxtx);
+        usbd_ep_write(dev, WEBUSB_TXD_EP, 0, 0);
+#endif
         return usbd_ack;
     default:
         return usbd_fail;
     }
 }
 
-void cdc_init_usbd(void) {
-    usbd_init(&libusb_udev, &usbd_hw, CDC_EP0_SIZE, ubuf, sizeof(ubuf));
-    usbd_reg_config(&libusb_udev, cdc_setconf);
-    usbd_reg_control(&libusb_udev, cdc_control);
-    usbd_reg_descr(&libusb_udev, cdc_getdesc);
+void usb_device_config_init(void) {
+    usbd_init(&libusb_udev, &usbd_hw, 8, ubuf, sizeof(ubuf));
```

Comment:
here macro is replaced with a constant value. make sure it is expected

Reply:
updated